Support mopidy on Darwin

Kaali opened the pull request:
###### Motivation for this change

I wanted to try out mopidy on Darwin.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Had to modify a bunch of gstreamer related packages to get them to work on macOS. There were some hard parts in gst-plugins-bad, which resulted in me disabling Cocoa  and mjpegtools support in it.

As a side-note I added mpg123 to gst-plugins-ugly to enable MP3 playback in mopidy.

This PR is related to issue https://github.com/NixOS/nixpkgs/issues/35322

As a global change, I replaced hardcoded SSL cert path with cacert. Not sure if that is the way it should be done.

Commit:
gst-plugins-bad: Add Darwin support

Had to remove mjpegtools and Cocoa support because of compile
problems.

Commit:
glib-networking: Fix Darwin support

Commit:
mopidy: Remove dbus-python on Darwin

Darwin doesn't have dbus.

Commit:
gst-plugins-ugly: Add mpg123 to enable mpg123audiodec in gstreamer

Adds support for MP3 playback.

LnL7 commented on pkgs/development/libraries/glib-networking/default.nix:
```diff
@@ -15,7 +15,9 @@ stdenv.mkDerivation rec {
   outputs = [ "out" "dev" ]; # to deal with propagatedBuildInputs
 
-  configureFlags = "--with-ca-certificates=/etc/ssl/certs/ca-certificates.crt";
+  configureFlags = "--with-ca-certificates=${cacert}/etc/ssl/certs/ca-bundle.crt";
```
This might have been intentional, not sure.

matthewbauer replied:
Yes see 55932c1beca26c7b5b7c259d95f6eb80644150a7 for reasoning. I'm not sure if anything is installed in /etc/ssl by default on macOS though.

Kaali replied:
As far as I know, macOS uses a patched OpenSSL, which looks for the keys in KeyChain. So there really is no certificates file. Homebrew handles this by generating OpenSSL certificate bundle from KeyChain on install.  Macports seems to just use curl ca-certs.

It seems like they are only updated when those packages are reinstalled. It might break the philosophy of Nix (as far as I understand it), but should there be a way to keep up an external state for this situation?

Kaali replied:
Found out a bigger issue for this: https://github.com/NixOS/nixpkgs/issues/8247

Maybe I'll just disable certificates for now. At least it will work for local files and non-secure remote files. Which really covers most of the music streams, I guess.

Commit:
glib-networking: Disable ca-certificates on Darwin

macOS does not have certificates at /etc/ssl/certs. cacerts package
has been deprecated. And there is a long standing issue at
https://github.com/NixOS/nixpkgs/issues/8247 for figuring out how to
handle certificates.

Disabling glib-networking ca-certificates on Darwin removes constant
warnings when opening remote connections.

LnL7 commented:
gnutls doesn't support NIX_SSL_CERT_FILE?

Kaali commented:
The configuration variable expects a path. I see that some packages have been patched to read the environment variable. Is that the suggested approach?

LnL7 commented:
Yes, with openssl NIX_SSL_CERT_FILE and SSL_CERT_FILE are use to override this at runtime making the default less important. The installer will point this to the default profile which should always contain cacert.

On darwin there's also `/etc/ssh/cert.pem` but I'm not sure how reliable that is, I've seen people running into issues with that. Alternatively any system installed with the nix installer will have a default profile, using `/nix/var/nix/profiles/default/etc/ssl/certs/ca-bundle.crt` will be equivalent to the default NIX_SSL_CERT_FILE. Perhaps that's the most reasonable default since it's still possible to override that by installing a custom cacert package.

LnL7 commented:
Note, fixing gnutls in this pr isn't neccecary but it's something we should probably look into.
